### PR TITLE
Accept it hunti

### PR DESCRIPTION
**Pull request by JuliuszMaretzky**
I messed a little in branches, but think it's ok anyway...

### CODE REVIEW

**Comment by huntipl** [comment]
1. give the PR better name expressing what the code is trying to do. You can keep it as the branch name, or call it something like "implemented changes doing xxx"

File: Seed.Tests/PlayerTests.cs [hunk 1]
```diff
@@ -704,7 +708,7 @@ public void ShouldDisplayOneOfCommentsFromListWhenRest()
             } while (player.Energy > 0);
             StringWriter swr = new StringWriter();
             Console.SetOut(swr);
-            List<string> RestSamples = new List<string>()
+            var RestSamples = new List<string>()
```
Comment (huntipl) [inline Seed.Tests/PlayerTests.cs:711]: restSamples

File: Seed/Service.cs [hunk 2]
```diff
@@ -494,12 +498,12 @@ public static void Command(Player player, World world)
 
                 foreach (var c in world.NPCs)
                 {
-                    if (typeof(IMove).IsAssignableFrom(c.GetType()))
+                    if (c is IMove)
                     {
-                        GetCharacterMove((IMove)c);
+                        MoveCharacter((IMove)c);
```
Comment (huntipl) [inline Seed/Service.cs]: you're doing double casting here.

```
if(c is IMove move)
{
MoveCharacter(move);
}
```

File: Seed/Service.cs [hunk 3]
```diff
@@ -494,12 +498,12 @@ public static void Command(Player player, World world)
...
                     }
 
-                    if (typeof(IAttack).IsAssignableFrom(c.GetType()))
+                    if (c is IAttack)
```
Comment (huntipl) [inline Seed/Service.cs:508]: same as above

**Review by huntipl — CHANGES_REQUESTED** [review]
required changes

**Comment by JuliuszMaretzky** [comment]
Required changes commited.

**Comment by JuliuszMaretzky** [comment]
I added new commits, please look at them.

File: Seed.Tests/PlayerTests.cs (hunk 1 above)
Comment (huntipl) [reply Seed.Tests/PlayerTests.cs:711]: still capital R

File: Seed/Characters/Player.cs [hunk 4]
```diff
@@ -466,7 +469,9 @@ public void Rest()
 
         public void Read(Book book)
         {
+            Console.ForegroundColor = ConsoleColor.White;
```
Comment (huntipl) [inline Seed/Characters/Player.cs:472]: why are you setting this to White to change it to grey in a second?

File: Seed/Characters/Player.cs [hunk 5]
```diff
@@ -476,7 +481,35 @@ public void Watch(Location location)
 
         public void Watch(Character character)
         {
+            Console.ForegroundColor = ConsoleColor.White;
             Service.DisplayLongString("\n" + character.Overview + "\n");
+
+            Console.Write($"{character.Name} jest ");
+            if(character.HP==character.MaxHP)
```
Comment (huntipl) [inline Seed/Characters/Player.cs:488]: could be refactored to a method isCharacterFullHP, to contain the logic within the character class

**Review by huntipl — COMMENTED** [review]
I would do some changes but it's better now so up to you

**Review by huntipl — APPROVED** [review]

**Pull request merged**